R/helper_add_fixed_drives.R

```diff
@@ -15,7 +15,7 @@ add_drive_results <- function(d) {
       posteam = dplyr::case_when(
         # on kickoffs the kicking team is the defteam but this should be swapped
         # in terms of this function if the kickoff is recovered
-        .data$own_kickoff_recovery == 1 ~ .data$defteam,
+        .data$own_kickoff_recovery == 1 | .data$fumble_lost == 1 ~ .data$defteam,
```

Comment:
This posteam fix should be added to `add_series_data()` as well

Reply:
I don't understand why that would be necessary- series automatically increments upon a new drive right?

Reply:
posteam is used for the drive result but only for touchdowns. Not 100% sure if it is necessary or not
https://github.com/mrcaseb/nflfastR/blob/1c81c26e02da5bb73cad01192c4909cffc46f92a/R/helper_add_series_data.R#L53-L54

Reply:
Ah so if a team fumbles on a kickoff and it's returned for a TD, we need to decide of the result should count as "Touchdown" or "Opp touchdown". I would say the 2nd?

Reply:
It should match the team that makes the PAT because `fixed_drive` won't change. So it should be `"Touchdown"`?

Reply:
A game with a fumbled kickoff TD for checking
```
g <- fast_scraper("2016_03_NYJ_KC")

g %>%
  # filter(qtr == 2) %>%
  filter(between(play_id, 1200, 1400)) %>%
  select(
    play_id, posteam, desc, fixed_drive, fixed_drive_result, series, series_result
  )
  play_id posteam desc                                                         fixed_drive fixed_drive_res~ series series_result
    <dbl> <chr>   <chr>                                                              <dbl> <chr>             <dbl> <chr>        
1    1210 KC      (7:47) 11-A.Smith pass short right to 87-T.Kelce to NYJ 4 f~           6 Field goal           18 Field goal   
2    1234 KC      (7:04) (Shotgun) 32-S.Ware right end to NYJ 9 for -5 yards ~           6 Field goal           18 Field goal   
3    1255 KC      (6:21) 11-A.Smith pass incomplete short right to 17-C.Conle~           6 Field goal           18 Field goal   
4    1277 KC      (6:15) 5-C.Santos 27 yard field goal is GOOD, Center-41-J.W~           6 Field goal           18 Field goal   
5    1296 NYJ     5-C.Santos kicks 61 yards from KC 35 to NYJ 4. 89-J.Marshal~           7 Touchdown            19 Touchdown    
6    1330 KC      5-C.Santos extra point is GOOD, Center-41-J.Winchester, Hol~           7 Touchdown            19 Touchdown    
7    1346 NYJ     5-C.Santos kicks 65 yards from KC 35 to NYJ 0. 89-J.Marshal~           8 Field goal           20 First down   
8    1379 NYJ     (5:54) (Shotgun) 14-R.Fitzpatrick pass incomplete deep midd~           8 Field goal           20 First down 
```